### DISCUSSION TITLE
Phaser in Qt QML

### DISCUSSION THREAD

**Discussion by jhay06** [photonstorm/phaser#6718] [Feature Requests]
Hi, i know this is far of being priority or may not ever become one of the piority itself, but maybe someone can consider adding Phaser in QML QT since JS is supported on QML but no HTML, window, document , navigator or any related on web . 

I tried but seems there so many web dependencies api used, so im checking if there is someone try to consider , if none its okay 
thank you .

**Comment by photonstorm** [comment]
Does QT support the Canvas tag? If not, this wouldn't really be possible, as we won't be able to render anything. If it does, then this is something we could probably look at.

**Reply by jhay06** [reply]
hi , Qt has its own Canvas QML ,  
https://doc.qt.io/qt-6/qml-qtquick-canvas.html 
im not into a web app , but a Qt native app , i believe this is possible but need to work out these document , window, navigator, HTMLCanvasElement , and other web related objects.  

but running js file in Qt is very well supported without web since Qt is using V8 engine as their JS Engine

**Reply by photonstorm** [reply]
Right, so it offers a sort of emulated Canvas API. All of the major calls are in there it appears, so it _should_ work with Phaser, if Phaser didn't do any of the other browser API tests. Like, it would work with a tiny subset of Phaser.

**Reply by jhay06** [reply]
yes ,it has major api for canvas , hope that phaserjs can support Qt  :)

**Reply by photonstorm** [reply]
Wouldn't it be easier to just use Qt WebView? https://doc.qt.io/qt-6/qtwebview-index.html

**Reply by jhay06** [reply]
yes it would be easier ,but it will be better if we can run it natively , anyways I'm still trying to port it but it has too many web specific element ,  also im still hoping that this will be added in future :)
